New cookies banner and page

tombye opened the pull request:
Updates the cookie banner and cookies page so the use of Google Analytics is based on user consent.

## Cookie banner
### If user has not accepted or rejected analytics:
![cookies_banner_no_consent](https://user-images.githubusercontent.com/87140/72088226-a38afc00-3301-11ea-8b87-972587832a43.png)
### If user has accepted analytics:
![cookie_banner_accepted](https://user-images.githubusercontent.com/87140/71728931-6a044d80-2e36-11ea-817b-2bae295a2b5b.png)
### If user has rejected analytics:
![cookie_banner_rejected](https://user-images.githubusercontent.com/87140/71728936-6cff3e00-2e36-11ea-9345-848d2ab10f6c.png)

Note: the banner will not appear if JS is not available.

### How it works
How this works is best explained by its JS tests.
```
    If user has made a choice to give their consent or not, the cookie banner should be hidden
    If user hasn't made a choice to give their consent or not
      - the cookie banner should show
      - no analytics should run
      - if the user clicks the button to accept analytics
        - the banner should confirm your choice and link to the cookies page as a way to change your mind
        - if the user clicks the 'hide' button, the banner should be hidden
        - the consent cookie should be set, with analytics set to 'true'
        - the analytics should be set up
      - if the user clicks the button to reject analytics
        - the banner should confirm your choice and link to the cookies page as a way to change your mind
        - if the user clicks the 'hide' button, the banner should be hidden
        - the consent cookie should be set, with analytics set to 'false'
        - the analytics should not be set up
```

Run these tests by:
`npx jest --config tests/javascripts/jest.config.js tests/javascripts/cookieMessage.test.js`

## Cookie page
You can now accept or reject analytics at any time using a form on the cookies page.
Like the cookies banner, the content for the whole page has been updated.
### Confirmation message
When you save your choice, you get a message confirming this:
#### Default:
![cookies_page_confirmation](https://user-images.githubusercontent.com/87140/71729047-aa63cb80-2e36-11ea-9fb3-d033e0b2765b.png)
#### If user came from another page to this one:
![cookies_page_confirmation_with_previous_page](https://user-images.githubusercontent.com/87140/71729052-adf75280-2e36-11ea-979b-2ba727f9d463.png)
#### Content for users without JS available
The section allowing you to choose settings for analytics requires JS (they are JS cookies) so has alternative content for users without JS available.

### How it works
```
    When the page loads
      - if user has not chosen to accept or reject analytics, the radios for making that choice should be set to unchecked
      - if analytics are accepted, the radio for 'accept analytics' should be set to checked
      - if analytics are rejected, the radio for 'reject analytics' should be set to checked
    When the 'Save cookie settings' button is clicked
      - if no selection is made, set consent to reject analytics
      - if a selection is made, save this as consent
      - the message confirming your choice
        - should be shown when the 'Save cookie settings' button is clicked
        - should include a link to the last page visited, if information on the referrer is available
      - analytics code
        - if user accepted analytics, the analytics code should initialise and register a pageview
        - if user rejected analytics, the analytics code should not run
```
Run these tests by:
npx jest --config tests/javascripts/jest.config.js tests/javascripts/cookieMessage.test.js

quis commented:
Summary of this afternoon’s discussion:
- hide button should have blue text and a 👆cursor
- banner shouldn’t be shown to users who don’t have Javascript
- messaging should be a bit more like this:
   <img width="1255" alt="Screenshot 2020-01-07 at 16 50 34" src="https://user-images.githubusercontent.com/355079/71914875-722ff600-3172-11ea-8290-e1818e39cb5f.png">

karlchillmaid commented on app/templates/components/cookie-banner.html:
```diff
@@ -0,0 +1,25 @@
+<div id="{{ id }}" class="notify-cookie-banner notify-cookie-banner__with-js" data-module="cookie-banner" role="region" aria-describedby="notify-cookie-banner__heading">
+  <div class="notify-cookie-banner__wrapper govuk-width-container">
+    <h2 class="notify-cookie-banner__heading govuk-heading-m" id="notify-cookie-banner__heading">Can we store analytics cookies on your device?</h2>
+    <p class="govuk-body">These help us understand how our website is being used.</p>
```
@tombye Maybe `These cookies` or `Analytics cookies` would read better?

tombye replied:
Now updated, in https://github.com/alphagov/notifications-admin/commit/d0453164547b0ebd3054ce793b961c8b7e98423c so marking this as resolved.

tombye commented:
I've made the changes described by @quis and @karlchillmaid in https://github.com/alphagov/notifications-admin/pull/3238/commits/d0453164547b0ebd3054ce793b961c8b7e98423c and updated the examples in the description to match.

I also added some (hidden) text to the buttons to help them make sense out of context (details in https://github.com/alphagov/notifications-admin/pull/3238/commits/40ac30a96da5b3147586c6929345161c75e19536).

karlchillmaid commented on app/templates/views/cookies.html:
```diff
@@ -37,21 +41,42 @@ <h2 class="heading-medium">Session cookies</h2>
+      Google is not allowed to use or share our analytics data with anyone.
+    </p>
+    <p>
+      Google Analytics cookies collect and store anonymised data about:
```
GOV.UK's updated content says:

>Google Analytics stores information about:
>
>- how you got to the site
>- the pages you visit on GOV.UK and how long you spend on them
>- what you click on while you’re visiting the site

Should we be saying exactly the same, or is this because they use some cookies that we don't?

tombye commented:
> Summary of this afternoon’s discussion:
> 
>     * hide button should have blue text and a 👆cursor
> 
>     * banner shouldn’t be shown to users who don’t have Javascript
> 
>     * messaging should be a bit more like this:
>       <img alt="Screenshot 2020-01-07 at 16 50 34" width="1255" src="https://user-images.githubusercontent.com/355079/71914875-722ff600-3172-11ea-8290-e1818e39cb5f.png">

I made these changes and updated the description to show it.
